%% PNAStmpl.tex
%% Template file to use for PNAS articles prepared in LaTeX
%% Version: Apr 14, 2008


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% BASIC CLASS FILE
%% PNAStwo for two column articles is called by default.
%% Uncomment PNASone for single column articles. One column class
%% and style files are available upon request from pnas@nas.edu.
%% (uncomment means get rid of the '%' in front of the command)

%\documentclass{pnasone}
\documentclass{pnastwo}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Changing position of text on physical page:
%% Since not all printers position
%% the printed page in the same place on the physical page,
%% you can change the position yourself here, if you need to:

% \advance\voffset -.5in % Minus dimension will raise the printed page on the
                         %  physical page; positive dimension will lower it.

%% You may set the dimension to the size that you need.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% OPTIONAL GRAPHICS STYLE FILE

%% Requires graphics style file (graphicx.sty), used for inserting
%% .eps files into LaTeX articles.
%% Note that inclusion of .eps files is for your reference only;
%% when submitting to PNAS please submit figures separately.

%% Type into the square brackets the name of the driver program
%% that you are using. If you don't know, try dvips, which is the
%% most common PC driver, or textures for the Mac. These are the options:

% [dvips], [xdvi], [dvipdf], [dvipdfm], [dvipdfmx], [pdftex], [dvipsone],
% [dviwindo], [emtex], [dviwin], [pctexps], [pctexwin], [pctexhp], [pctex32],
% [truetex], [tcidvi], [vtex], [oztex], [textures], [xetex]

%\usepackage[dvips]{graphicx}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% OPTIONAL POSTSCRIPT FONT FILES

%% PostScript font files: You may need to edit the PNASoneF.sty
%% or PNAStwoF.sty file to make the font names match those on your system.
%% Alternatively, you can leave the font style file commands commented out
%% and typeset your article using the default Computer Modern
%% fonts (recommended). If accepted, your article will be typeset
%% at PNAS using PostScript fonts.

% Choose PNASoneF for one column; PNAStwoF for two column:
%\usepackage{PNASoneF}
\usepackage{PNAStwoF}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% ADDITIONAL OPTIONAL STYLE FILES

%% The AMS math files are commonly used to gain access to useful features
%% like extended math fonts and math commands.

\usepackage{amssymb,amsfonts,amsmath}

\graphicspath{ {paper_figures2/} }

%\usepackage{tikz}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% OPTIONAL MACRO FILES
%% Insert self-defined macros here.
%% \newcommand definitions are recommended; \def definitions are supported

%\newcommand{\mfrac}[2]{\frac{\displaystyle #1}{\displaystyle #2}}
%\def\s{\sigma}


\newcommand{\SI}[0]{{\it SI Materials and Methods}}
\newcommand{\fig}[0]{Fig.}


\makeatletter
\newcommand{\customlabel}[2]{%
\protected@write \@auxout {}{\string \newlabel {#1}{{#2}{}}}}
\makeatother

\customlabel{fig:DV_view}{S1}
\customlabel{fig:membrane_curve}{S2}
\customlabel{fig:raw_data3}{S3}
\customlabel{fig:1d_example}{S4}
\customlabel{fig:spheres}{S5}
\customlabel{fig:eigenvalues}{S6}
\customlabel{fig:PCA_data1}{S7}
\customlabel{fig:PCA_data2}{S8}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Don't type in anything in the following section:
%%%%%%%%%%%%
%% For PNAS Only:
\contributor{Submitted to Proceedings
of the National Academy of Sciences of the United States of America}
\url{www.pnas.org/cgi/doi/10.1073/pnas.0709640104}
\copyrightyear{2008}
\issuedate{Issue Date}
\volume{Volume}
\issuenumber{Issue Number}
%%%%%%%%%%%%

\begin{document}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%% For titles, only capitalize the first letter
%% \title{Almost sharp fronts for the surface quasi-geostrophic equation}

\title{Temporal ordering and registration of images in studies of developmental dynamics}

%% Enter authors via the \author command.
%% Use \affil to define affiliations.
%% (Leave no spaces between author name and \affil command)

%% Note that the \thanks{} command has been disabled in favor of
%% a generic, reserved space for PNAS publication footnotes.

%% \author{<author name>
%% \affil{<number>}{<Institution>}} One number for each institution.
%% The same number should be used for authors that
%% are affiliated with the same institution, after the first time
%% only the number is needed, ie, \affil{number}{text}, \affil{number}{}
%% Then, before last author ...
%% \and
%% \author{<author name>
%% \affil{<number>}{}}

%% For example, assuming Garcia and Sonnery are both affiliated with
%% Universidad de Murcia:
%% \author{Roberta Graff\affil{1}{University of Cambridge, Cambridge,
%% United Kingdom},
%% Javier de Ruiz Garcia\affil{2}{Universidad de Murcia, Bioquimica y Biologia
%% Molecular, Murcia, Spain}, \and Franklin Sonnery\affil{2}{}}

\author{Carmeline~J.~Dsilva\affil{1}{Department of Chemical and Biological Engineering, Princeton University, Princeton, New Jersey, USA},
Bomyi~Lim\affil{1}{},
Thomas~J.~Levario\affil{2}{School of Chemical and Biomolecular Engineering, Georgia Institute of Technology, Atlanta, Georgia, USA},
Hang~Lu\affil{2}{},
Amit~Singer\affil{3}{Department of Mathematics, Princeton University, Princeton, New Jersey, USA} \affil{4}{Program in Applied and Computational Mathematics, Princeton University, Princeton, New Jersey, USA},
Stanislav~Y.~Shvartsman\affil{1}{} \affil{5}{Lewis-Sigler Institute for Integrative Genomics, Princeton University, Princeton, New Jersey, USA},
\and
Ioannis~G.~Kevrekidis\affil{1}{} \affil{4}{Program in Applied and Computational Mathematics, Princeton University, Princeton, New Jersey, USA}}

\contributor{Submitted to Proceedings of the National Academy of Sciences
of the United States of America}

%\significancetext{Cross-sectional imaging studies are commonly used to study developmental dynamics in multiple organisms. 
%%
%In these studies, a representative developmental trajectory is reconstructed from snapshots of different embryos, each of which has been arrested at a different point of its development. 
%%
%One of the first steps in this reconstruction involves registration and temporal ordering of images. 
%%
%Here we show that both of these tasks can be automated and combined in a single step using recent advances in data mining. 
%%
%To illustrate this approach, we temporally order and register several data sets collected in cross-sectional studies of cell signaling in the early fruit fly embryo.}

%% The \maketitle command is necessary to build the title page.
\maketitle

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{article}

\begin{abstract}

Imaging studies provide unique insights into the dynamics of developmental processes. 
%
Many experiments are based on a cross-sectional design, where developmental progress is arrested in a population of embryos, each of which is at a different point along its developmental trajectory. 
%
The goal is then to temporally order the data to reconstruct representative developmental dynamics from snapshots provided by the collection of embryos.
%
Images of different biological samples must first be registered before they can be temporally ordered.
%
When such data sets are large, noisy, and/or if the developmental changes are subtle, these tasks can be difficult to perform manually.
%
We present an automated approach to simultaneously register and temporally order data from such cross-sectional studies.
%
The approach is based on vector diffusion maps, a broadly applicable manifold learning technique that does not require {\it a priori} knowledge of image features or a parametric model of the developmental dynamics.
%
We use this method to extract developmental trajectories from a cross-sectional imaging study of cell signaling during dorsoventral patterning of the {\it Drosophila} embryo.
%
We show that the algorithm recovers the correct orientation and temporal order in a data set where they are independently known, reconstructs an average developmental trajectory when such information is unavailable, and can detect distinct trajectories in a heterogeneous population of embryos.

\end{abstract}

%% When adding keywords, separate each term with a straight line: |
\keywords{temporal ordering | image registration | vector diffusion maps}

\section{Significance}
Cross-sectional imaging studies are commonly used to study developmental dynamics in multiple organisms. 
%
In these studies, a representative developmental trajectory is reconstructed from snapshots of different embryos, each of which has been arrested at a different point of its development. 
%
One of the first steps in this reconstruction involves registration and temporal ordering of images. 
%
Here we show that both of these tasks can be automated and combined in a single step using recent advances in data mining. 
%
To illustrate this approach, we temporally order and register several data sets collected in cross-sectional studies of cell signaling in the early fruit fly embryo.
\vspace{0.5cm}

%% Optional for entering abbreviations, separate the abbreviation from
%% its definition with a comma, separate each pair with a semicolon:
%% for example:
%% \abbreviations{SAM, self-assembled monolayer; OTS,
%% octadecyltrichlorosilane}

% \abbreviations{}

%% The first letter of the article should be drop cap: \dropcap{}
%\dropcap{I}n this article we study the evolution of ''almost-sharp'' fronts

%% Enter the text of your article beginning here and ending before
%% \begin{acknowledgements}
%% Section head commands for your reference:
%% \section{}
%% \subsection{}
%% \subsubsection{}


\dropcap{E}xperimental studies of developmental dynamics fall in two broadly defined categories: longitudinal and cross-sectional \cite{diggle2002analysis}.
%
In longitudinal studies, developmental progress is monitored over time in the same embryo \cite{roelens2013live, keller2013imaging}.
%
In a cross-sectional study, developmental dynamics is reconstructed from multiple embryos, each of which contributes only a snapshot of a chemical or morphological process along its developmental trajectory \cite{jaeger2004dynamic, peter2011gene, fowlkes2008quantitative}.
%
Here we focus on cross-sectional studies, which have a time-honored history and still present the only option for most organisms.
%
In a typical cross-sectional study, a group of embryos is fixed using a procedure that arrests their development, and stained with chemicals that visualize a handful of cellular processes.
%
Fixed embryos are then imaged using any given number of microscopy techniques.
%
Importantly, the ``age'' of any given embryo arrested in its development is not quantitatively known; typically what is known is
a certain time window to which a collection of embryos belongs \cite{ng2012large, richardson2014emage, castro2009automatic}.
%
In order to recover the developmental dynamics from such data sets, snapshots of different embryos must first be spatially aligned or {\em registered} to factor out the relevant geometric symmetries (e.g., translations and rotations), and then ordered in time.
%
We show how recently developed dimensionality reduction algorithms can automate {\it and combine} both of these tasks.

Temporal ordering and registration of images can be done manually
when the number of images is small and the differences between them are visually apparent. 
%
\fig~\ref{fig:fish} shows a caricature of fish development which illustrates the processes of growth and patterning.
%
In this case, temporal ordering can be accomplished by arranging the fish by size, which is monotonic with the developmental progress.
%
Image registration is based on obvious morphological landmarks, such as the positions of the head and the fins.
%
In contrast to this example, real data poses nontrivial challenges for both registration and temporal ordering.
%
In general, the landmarks needed for registration, as well as the attributes which can be used to order the data, are not known {\it a priori}.
%
Additional challenges arise from embryo-to-embryo variability, sample size, and measurement noise.

\begin{figure}[t]
\includegraphics{fig1}
\caption{Caricature illustrating the tasks of image registration and temporal ordering. {\xfigtextfontit (A)} Images of ``samples'', each in a different orientation and a different stage of development. {\xfigtextfontit (B)} Registered and ordered samples. For this caricature, the registration and ordering is straightforward because the data set is small, the landmarks are visually apparent, and the developmental changes are easy to recognize.}
%\label{fig:fish}
\customlabel{fig:fish}{1}
\customlabel{subfig:fish_unordered}{\ref{fig:fish}{\it A}}
\customlabel{subfig:fish_ordered}{\ref{fig:fish}{\it B}}
\end{figure}

Image registration has been studied in a variety of contexts \cite{zitova2003image, rowley1998rotation, hajnal2010medical, greenspan1994rotation, zhao2003face}, 
most of which require the definition and identification of appropriate landmarks.
%
In contrast, our approach requires no such information. 
%
It is based on a manifold learning algorithm (vector diffusion maps, VDM \cite{singer2012vector}) which simultaneously addresses the problems of registration and temporal ordering. 
%
This algorithm is one of several influential nonlinear dimensionality reduction techniques that have been developed over the past decade \cite{Belkin2003, coifman2005geometric, coifman2006geometric, tenenbaum2000global, roweis2000nonlinear}. 
%
These techniques has been used in
applications ranging from analysis of cryo-electron microscopy (cryo-EM) images of individual molecules  \cite{zhao2014rotationally, singer2011viewing} to face recognition \cite{lafon2006data}, and are sufficiently general to be applicable to a wide variety of biological imaging studies.
%
Here, the vector diffusion maps algorithm is adapted for the analysis of images from cross-sectional studies of developmental dynamics, with the main objective of revealing stereotypic developmental trajectories.
%
To illustrate our approach, we analyze three data sets from a study of {\it Drosophila} embryogenesis, one of the best experimental models for studies of developmental dynamics \cite{jaeger2012drosophila}.


\section{Results}

\subsection{Vector diffusion maps for registration and temporal ordering}

Vector diffusion maps \cite{singer2012vector} is a manifold learning
technique developed for data sets which contain two types of sources of variability:
geometric symmetries (such as translations and rotations of the images due to the experimental setup) which one would like to factor out,
and ``additional" directions of variability (such as temporal dynamics) which one wants to uncover.
%
Vector diffusion maps combine two algorithms, {\em angular synchronization} \cite{singer2011angular} for image registration and {\em diffusion maps} \cite{coifman2005geometric} for extracting intrinsic low-dimensional structure in data, into a single computation.
%
We will use the algorithm to register images of {\it Drosophila} embryos with respect to rotations and translations, as well as uncover the main direction of variability {\it after} removing translational and rotational symmetries.
%
We assume that the main direction of variability in these images is parameterized by the developmental time of each embryo, so that uncovering this direction will allow us to recover the developmental dynamics.

Angular synchronization uses pairwise alignment information to register a set of images in a globally consistent way.
%
A schematic illustration of angular synchronization is shown in \fig~\ref{subfig:synch1}, where each image is represented as a vector, and the goal is to align the set of vectors.
%
We first compute the angles needed to align pairs of vectors (or images).  
%
In general, this requires no template function \cite{ahuja2007template} or image landmarks \cite{ian1998statistical}.
%
Using the alignment angles between all pairs of vectors, angular synchronization finds the set of rotation angles (one angle for each vector) that is most consistent with {\it all} pairwise measurements (see \SI); this is illustrated in \fig~\ref{subfig:synch2}.
%
In this schematic, registration via angular synchronization is trivial, as the pairwise measurements contain no noise.
%
However, the algorithm can successfully register data sets even when many of the pairwise measurements are inaccurate \cite{singer2011angular}.
%

%
After removing variability due to translations and rotations, the developmental dynamics may be revealed by ordering the data along the one-dimensional manifold that parameterizes most of the remaining variability in the data.
%
Such a manifold can be discovered using diffusion maps \cite{coifman2005geometric}, a nonlinear dimensionality reduction technique that uncovers an intrinsic parametrization of data that lies on a low-dimensional, perhaps nonlinear, manifold in high-dimensional space.
%
The idea is illustrated in \fig~\ref{subfig:dmaps1}, where the data are two-dimensional points which lie on a one-dimensional nonlinear curve.
%
We use {\it local} information about the data to find a parametrization  which respects the underlying manifold geometry: we want points which are close in high-dimensional space (e.g., images which look similar) to be close in our parametrization.
%
This idea of locality is denoted by the edges in \fig~\ref{subfig:dmaps1}.
%
Data points which are close are connected by dark edges, and clearly, the dark edges are more ``informative" about the low-dimensional structure of the data.
%
The color in \fig~\ref{subfig:dmaps2} depicts the one-dimensional parametrization or ordering of the data that we can detect visually.
%
In our working examples, each data point will be of much higher dimension (e.g., a pixelated image), and so we cannot extract this low-dimensional structure visually.
%
Instead, we will use diffusion maps to automatically uncover a parametrization of our high-dimensional data (see \SI).
%
Diffusion maps generalizes directly from one-dimensional nonlinear curves to higher-dimensional manifolds.
%
Thus, when our experiments have additional sources of variability (such as distinct populations of embryos) in addition to developmental dynamics, the algorithm can provide an informative organization of our data which highlights both of these variability sources.

\begin{figure}[t]
\includegraphics{fig2}
\caption{Schematic illustrating angular synchronization and diffusion maps. {\xfigtextfontit (A)} Set of vectors, each in a different orientation. The pairwise alignment angles are indicated. {\xfigtextfontit (B)} The vectors from {\xfigtextfontit A}, each rotated about their midpoint so that the set is globally aligned. Note that the chosen rotation angles are consistent with the pairwise alignments in {\xfigtextfontit A}: the differences between a pair of angles in {\xfigtextfontit B} is the same as the pairwise angle in {\xfigtextfontit A}. {\xfigtextfontit (C)} Data points (in black) which lie on a one-dimensional nonlinear curve in two dimensions. Each pair of points is connected by an edge, and the edge weight is related to the Euclidean distance between the points through the diffusion kernel (see {\xfigtextfontit \SI}), so that close data points are connected by darker (``stronger'') edges. {\xfigtextfontit (D)} The data in {\xfigtextfontit C}, colored by the first (non-trivial) eigenvector from the diffusion map computational procedure. The color intensity is monotonic with the perceived curve arclength, thus parameterizing the curve.}
%\label{fig:schematics}
\customlabel{fig:schematics}{2}
\customlabel{subfig:synch1}{\ref{fig:schematics}{\it A}}
\customlabel{subfig:synch2}{\ref{fig:schematics}{\it B}}
\customlabel{subfig:dmaps1}{\ref{fig:schematics}{\it C}}
\customlabel{subfig:dmaps2}{\ref{fig:schematics}{\it D}}
\end{figure}


\begin{figure*}[t]
\includegraphics{fig3}
\caption{{\xfigtextfontit Drosophila} embryos during cellularization. {\xfigtextfontit (A)} Images of embryos collected during cellularization. Each image is of a different embryo arrested at a different developmental time, and in a different rotational and translational orientation. {\xfigtextfontit (B)} Images from {\xfigtextfontit A}, now registered and ordered using vector diffusion maps. The images are ordered from left to right and from top to bottom, so that the first image appears in the top left position of the array, and the last image appears in the bottom right. The dorsal side of each embryo now appears at the top of each image, and the ventral side appears at the bottom.}
\customlabel{fig:movie}{3}
\customlabel{subfig:movie_unregistered_unordered}{\ref{fig:movie}{\it A}}
\customlabel{subfig:movie_registered_ordered}{\ref{fig:movie}{\it B}}
\customlabel{subfig:movie_angle_corr}{\ref{fig:movie}{\it C}}
\customlabel{subfig:movie_rank_corr}{\ref{fig:movie}{\it D}}
\customlabel{subfig:movie_error_table}{\ref{fig:movie}{\it E}}
\customlabel{subfig:movie_bootstrap}{\ref{fig:movie}{\it F}}
\end{figure*}


\subsection{Method validation using a live movie}

As a first illustration, we analyze a data set where the correct registration and temporal order is known {\it a priori}.
%
This data set is obtained by live imaging of a single {\it Drosophila} embryo during the thirty minute time interval spanning late cellularization through gastrulation.
%
During this time window the ventral furrow is formed, where the ventral side buckles towards the center of the embryo, bringing in the future muscle cells and eventually forming a characteristic ``omega'' shape.
%
Germband extension then causes cells from the ventral side to move towards the posterior pole of the embryo, and then wrap around to the dorsal side \cite{leptin2005gastrulation}.
%
At the end of this process, cells which were originally on the ventral and posterior side of the embryo find themselves on the dorsal side, causing similar ``omega'' shaped patterns to appear on the dorsal side.

The movie shows an optical cross-section of a vertically oriented developing embryo, where the nuclei are labeled by DAPI, a DNA stain.
%
The movie contains $42$ frames, taken at $30$~sec time intervals;
\fig~\ref{subfig:movie_unregistered_unordered} shows selected frames from this movie. 
%
The gray nuclear signal has been blurred to obtain a smoother images.
%
Each frame has been arbitrarily rotated, and the order of the frames has been scrambled.

We used vector diffusion maps to register and order the images. 
%
\fig~\ref{subfig:movie_registered_ordered} shows the same selected frames, now registered and ordered.
%
The real time for each frame is indicated.
%
\fig~\ref{subfig:movie_angle_corr} and \fig~\ref{subfig:movie_rank_corr}  shows the correlations between the recovered and true angles and ranks for each of the frames.
%
Both the angles and the ranks are recovered to a high degree of accuracy.

To assess the robustness of our proposed methodology, we repeated this procedure with four additional movies.
%
The results are shown in \fig~\ref{subfig:movie_error_table}. 
%
The errors in the recovered angles are all very low, and the rank correlation coefficients are consistently close to 1, indicating that our methodology can reproducably order data of this type.

To assess how the quality of our results depends on the number of data points, we used bootstrap sampling. 
%
\fig~\ref{subfig:movie_bootstrap} shows the average rank correlation coefficient as a function of the number of images in the data set.
%
The data sets were sampled (with replacement) from the original $40$-frame movie.
%
There is a steady increase in the accuracy of the recovered orderings with an increasing number of images.
%
The recovered angles had errors of less than $10^{\circ}$, and the magnitude of the errors did not vary significantly with the size of the data set.


\subsection{Images of gastrulating embryos}

We will now demonstrate the utility of our methodogies on a data set where the correct ordering is not known {\em a priori}.
%
\fig~\ref{subfig:fixed_images_unregistered_unordered} shows a set of $126$ images, which cover a thirty minute time interval spanning late cellularization through gastrulation.
%
Each image is an optical cross-section of a {\em different} vertically oriented embryo fixed at a different (and unknown) developmental time.
%
The nuclei (gray) are again labeled with a DNA marker.
%
Embryos were stained with the antibody that recognizes Twist (Twi, shown in green), a transcription factor which specifies the cells of the future muscle tissue.
%
Another signal is provided by the phosphorylated form of the extracellular signal regulated kinase (dpERK, red in the figure), an enzyme that specifies a subset of neuronal cells.

\begin{figure*}[t]
\includegraphics{fig4}
\caption{{\xfigtextfontit Drosophila} embryos during gastrulation. {\xfigtextfontit (A)} Images of gastrulating embryos. Each image is of a different embryo arrested at a different developmental time, and in a different translation and rotational orientation. {\xfigtextfontit (B)} Data from {\xfigtextfontit A}, registered and ordered using vector diffusion maps. The images are ordered from left to right and from top to bottom, so that the first image appears in the top left position of the array, and the last image appears in the bottom right. {\xfigtextfontit (C)} A representative ``developmental path'' from local averaging of the images. Each image is a (Gaussian-weighted) average of the images in {\xfigtextfontit B}. }
\customlabel{fig:data2}{4}
\customlabel{subfig:fixed_images_unregistered_unordered}{\ref{fig:data2}{\it A}}
\customlabel{subfig:fixed_images_registered_ordered}{\ref{fig:data2}{\it B}}
\customlabel{subfig:fixed_images_average_trajectory}{\ref{fig:data2}{\it C}}
\end{figure*}

FIX!!

\fig~\ref{subfig:ordered_data2} shows the images registered and ordered using vector diffusion maps \cite{singer2012vector}, 
and visual inspection confirms this ordering is consistent with the known developmental dynamics outlined above.
%
However, assimilating the dynamics even from this concatenation of images is difficult for two reasons, even after registration and ordering.
%
First, the sheer number of images makes visual processing of the entire data set nontrivial;
and second, the ordered data is not entirely smooth due to inter-embryo variability.
%
To highlight the developmental dynamics revealed by vector diffusion maps, we applied a Gaussian filter to the trajectory of registered and ordered images to obtain an averaged developmental path (see \SI and  \cite{kemelmacher2011exploring}).  
%
Sequential snapshots of this averaged trajectory, shown in \fig~\ref{subfig:averaged_data2}, serve as a summary of the stereotypic developmental dynamics reconstructed from cross-sectional data.
%
Thus, vector diffusion maps can accomplish the tasks presented in the caricature in \fig~\ref{fig:fish}, even in the absence of information about image landmarks (e.g., fins) and without {\it a priori} knowledge of developmental dynamics (e.g., correlation of age with body size).


%
dpERK appears as two lateral peaks at the ventral side of the embryo.
%
These two peaks merge together during invagination, eventually forming (together with Twi) a characteristic ``omega'' shape.
%
Germband extension then causes cells from the ventral side to move towards the posterior pole of the embryo, and then wrap around to the dorsal side \cite{leptin2005gastrulation}.
%
At the end of this process, cells which were originally on the ventral and posterior side of the embryo find themselves on the dorsal side, causing similar ``omega'' shaped patterns to appear on the dorsal side; these patterns are most readily seen in the last image of \fig~\ref{subfig:fixed_images_average_trajectory}.

\section{Discussion}

\begin{itemize}
\item High--throughput imaging is becoming increasingly common
\item We are/wil reach the point where imaging data is being produced more rapidly than it can be analyzed by hand (similar to microarray data)
\item We need a way to automate the organization of such data
\item Our algorithms register, which is important for many imaging applications
\item In general our algorithms find good coordinates/descriptors for the data
\item We focus on uncovering a single coordinate that can be used to order data
\item in general, can uncover many coordinates; use coordinates to do other data processing tasks (clustering, model fitting, etc.)

\end{itemize}
We presented a unified approach to temporal ordering and registration of images in cross-sectional studies of developmental dynamics. 
%
To the best of our knowledge, algorithmic approaches to these two tasks have been explored largely independently of each other. 
%
In particular, temporal ordering of imaging data sets was done with a significant amount of human supervision and using registered images as a starting point \cite{yuan2014automated, surkova2008characterization}.  
%
In parallel, ordering of large-scale cross-sectional data was done in the context of molecular profiling studies, in which data are vectors describing the amounts of different chemical species \cite{anavy2014blind, trapnell2014dynamics, gupta2008extracting}. 
%
Automated temporal ordering of such data sets was accomplished by first projecting the data onto a low-dimensional subspace spanned by the leading principal or independent components, and then solving a traveling salesman problem or constructing a minimum spanning tree on the projected data to order multiple snapshots. 
%
The approach presented here is different because it addresses image registration and ordering in a single step and orders the data without first projecting them on a low-dimensional subspace; vector diffusion maps can also order images when global registration of the entire data set is topologically impossible \cite{zhao2014rotationally}.
%
The task of image registration has been widely studied \cite{zitova2003image}, for applications such as face recognition \cite{rowley1998rotation}, medical image registration \cite{hajnal2010medical}, and texture classification \cite{greenspan1994rotation}.
%
In contrast to most of the existing approaches to image registration which rely on the knowledge about some landmarks in the data \cite{ian1998statistical} (such as the eyes in face recognition applications \cite{zhao2003face}), algorithms based on angular synchronization can register images even in the absence of such information. 

We are approaching the time when the rate of imaging data collection will surpass the rate of manual image analysis. 
%
We therefore require automated methodologies to organize such large data sets; these methodogies should be sufficiently general to address a wide variety of applications. 
%
Vector diffusion maps allows us to automatically register images, which is an essential task for many applications.
%
Simultaneously the algorithm provides us with a parameterization or coordinates for each image.
%
In the examples presented here, we have focued on recovering a single coordinate, and using this coordinate to order the images, effectively ordering them in time.
%
In general, we can recover an arbitrariy number of coordinates, and use this parametersization to do any of the typical data analysis tasks, such as outlier detection, clustering, and model fitting. 


We used vector diffusion maps to analyze three data sets, collected specifically for this study. 
%
In the first example, correct orientation and temporal order of images was known {\it a priori}, providing a test case for the algorithm in which both of these features were accurately recovered. 
%
For this particular data set, a simpler approach would have sufficed;
because of the relative simplicity of developmental dynamics (monotonic increase in the amplitude of a localized signaling pattern and small tissue deformation), temporal order could have been inferred by projection of registered images on the first principal component of the data set (see \fig~\ref{fig:PCA_data1}). 
%
This approach was not sufficient for the second data set (see \fig~\ref{fig:PCA_data2}), collected from embryos during the time window where both the tissue morphology and signaling patterns are rapidly changing.  
%
Vector diffusion maps could readily extract a nontrivial spatiotemporal trajectory. 
%
Finally, the algorithm was applied to a heterogeneous data set, composed of images from embryos of different genotypes. 
%
In this case, low-dimensional projection of the data revealed distinct paths which correlated with the developmental trajectories of the mutant and wild type populations. 

Previously, angular synchronization and vector diffusion maps have been used to reconstruct molecular shapes from cryo-electron microscopy images \cite{singer2012vector, zhao2014rotationally, singer2011viewing}.
%
Because of high levels of instrument noise in these data, thousands of images were needed for successful shape reconstruction. 
%
Based on the presented results, we expect that much smaller data sets may be sufficient for successful reconstruction of developmental trajectories. 
%
Here, we have used vector diffusion maps to register the images with respect to translations and rotations; however, it is conceivable that one could factor out other sources of variability, such as reflections \cite{singer2012vector, goemans1995improved, bandeira2013cheeger} or dilations.

For the data sets analyzed in this work, the Euclidean norm between the pixelated images was informative for identifying similar data points. 
%
For more complex data sets, for instance when the imaged samples are significantly deformed, working with raw images might be more challenging.   
%
In this case, one might choose to operate not on the images directly, but on those functions of the images that are invariant to symmetries and insensitive to small deformations.  
%
This is the basis of the recently developed scattering transform \cite{mallat2012group}, as well as the bispectrum-based steerable PCA algorithms in \cite{zhao2014rotationally}. 
%
Such methods circumvent the need for registration by using invariant features of the images in the data analysis.
%
Another direction for future work is related to the joint analysis of data sets provided by different imaging approaches, such as merging live imaging data of tissue morphogenesis with snapshots of cell signaling and gene expression from fixed embryos \cite{krzic2012multiview, ichikawa2014live, rubel2010coupling}. 
%
Given the rapidly increasing volumes of imaging data from studies of multiple developmental systems, we expect that dimensionality reduction approaches discussed in this work will be increasingly useful for biologists and motivate future applications and algorithmic advances. 
  
  
  \begin{itemize}
  \item Discuss parameter robustness/sensitity/tuning
  \item timing/scaling
  \item mention code
  \item SNR? Add Gaussian Noise?
  \end{itemize}

%% == end of paper:

%% Optional Materials and Methods Section
%% The Materials and Methods section header will be added automatically.

%% Enter any subheads and the Materials and Methods text below.


\begin{materials}

\section{Fly strain and whole-mount immunostaining}
%
Oregon-R (OreR) and spaghetti squash (Sqh)-GFP were used as wild type strains. 
%
{\it Sna\textsuperscript{\it IIG05}/Cyo hb-lacZ} flies were used to study mutant embryos with one peak of dpERK activation.  
%
Embryos were collected and fixed at 22$^\circ$C. 
%
Monoclonal rabbit anti-dpERK (1:100, Cell signaling), mouse anti-Dorsal (1:100, Developmental Studies Hybridoma Bank), rat anti-Twist (1:500, a gift from Eric Wieschaus), and mouse anti-beta-galactosidase (1:500, DSHB) were used to stain proteins of interest. DAPI (1:10,000, Vector Laboratories) was used to visualize nuclei, and Alexa Fluors (1:500, Invitrogen) were used as secondary antibodies. 

\section{Microscopy}
%
Nikon A1-RS scanning confocal microscope, and the Nikon 60X Plan-Apo oil objective was used to image embryos. Embryos were collected, stained, and imaged together under the same microscope setting. End-on imaging was performed by using the microfluidics device described previously. Images were collected at the focal plane $\sim$90~$\mathrm{\mu m}$ from the posterior pole of an embryo (see \fig~\ref{fig:DV_view}). 

\section{Development time during cellularization} 
%
The progression of membrane invagination during cellularization was used to assign time point to each embryo in the third hour of development. Time-lapse imaging of Sqh-GFP embryo was used to generate the calibration curve that relates the membrane ingression length and the developmental time. Time-lapsing imaging of 7 embryos was used to construct the calibration curve (\fig~\ref{fig:membrane_curve}B). The rate at which membrane invaginates was highly reproducible from embryo to embryo. In the experiments with fixed embryos, we measured the membrane length by co-staining embryos with a membrane marker (Sqh-GFP) or by taking phase-contrast images (\fig~\ref{fig:membrane_curve}A). We used the correlation curve to assign each embryo to a time window of 2--3~minutes. 

\end{materials}


%% Optional Appendix or Appendices
%% \appendix Appendix text...
%% or, for appendix with title, use square brackets:
%% \appendix[Appendix Title]


\begin{acknowledgments}
The authors thank Adam Finkelstein,  Thomas Funkhouser, and John Storey for helpful discussions. 
%
C.J.D. was supported by the Department of Energy Computational Science Graduate Fellowship (CSGF), grant number DE-FG02-97ER25308.
%
B.L. and S.Y.S. were supported by the National Institutes of Health Grant R01GM086537. 
%
T.J.L. and H.L. were supported by the National Science Foundation Grant Emerging Frontiers in Research and Innovation (EFRI) 1136913.
%
A.S. was supported by the Air Force Office of Scientific Research Grant
FA9550-12-1-0317.
%
I.G.K. was supported by the National Science Foundation (CS\&E program).
\end{acknowledgments}

%% PNAS does not support submission of supporting .tex files such as BibTeX.
%% Instead all references must be included in the article .tex document.
%% If you currently use BibTeX, your bibliography is formed because the
%% command \verb+\bibliography{}+ brings the <filename>.bbl file into your
%% .tex document. To conform to PNAS requirements, copy the reference listings
%% from your .bbl file and add them to the article .tex file, using the
%% bibliography environment described above.

%%  Contact pnas@nas.edu if you need assistance with your
%%  bibliography.

% Sample bibliography item in PNAS format:
%% \bibitem{in-text reference} comma-separated author names up to 5,
%% for more than 5 authors use first author last name et al. (year published)
%% article title  {\it Journal Name} volume #: start page-end page.
%% ie,
% \bibitem{Neuhaus} Neuhaus J-M, Sitcher L, Meins F, Jr, Boller T (1991)
% A short C-terminal sequence is necessary and sufficient for the
% targeting of chitinases to the plant vacuole.
% {\it Proc Natl Acad Sci USA} 88:10362-10366.


%% Enter the largest bibliography number in the facing curly brackets
%% following \begin{thebibliography}

\bibliographystyle{pnas2}
\bibliography{../image_analysis_paper/background_reading/references,../../references/references}

\end{article}

\end{document}




