%% PNAStmpl.tex
%% Template file to use for PNAS articles prepared in LaTeX
%% Version: Apr 14, 2008


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% BASIC CLASS FILE
%% PNAStwo for two column articles is called by default.
%% Uncomment PNASone for single column articles. One column class
%% and style files are available upon request from pnas@nas.edu.
%% (uncomment means get rid of the '%' in front of the command)

%\documentclass{pnasone}
\documentclass{pnastwo}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Changing position of text on physical page:
%% Since not all printers position
%% the printed page in the same place on the physical page,
%% you can change the position yourself here, if you need to:

% \advance\voffset -.5in % Minus dimension will raise the printed page on the
                         %  physical page; positive dimension will lower it.

%% You may set the dimension to the size that you need.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% OPTIONAL GRAPHICS STYLE FILE

%% Requires graphics style file (graphicx.sty), used for inserting
%% .eps files into LaTeX articles.
%% Note that inclusion of .eps files is for your reference only;
%% when submitting to PNAS please submit figures separately.

%% Type into the square brackets the name of the driver program
%% that you are using. If you don't know, try dvips, which is the
%% most common PC driver, or textures for the Mac. These are the options:

% [dvips], [xdvi], [dvipdf], [dvipdfm], [dvipdfmx], [pdftex], [dvipsone],
% [dviwindo], [emtex], [dviwin], [pctexps], [pctexwin], [pctexhp], [pctex32],
% [truetex], [tcidvi], [vtex], [oztex], [textures], [xetex]

%\usepackage[dvips]{graphicx}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% OPTIONAL POSTSCRIPT FONT FILES

%% PostScript font files: You may need to edit the PNASoneF.sty
%% or PNAStwoF.sty file to make the font names match those on your system.
%% Alternatively, you can leave the font style file commands commented out
%% and typeset your article using the default Computer Modern
%% fonts (recommended). If accepted, your article will be typeset
%% at PNAS using PostScript fonts.

% Choose PNASoneF for one column; PNAStwoF for two column:
%\usepackage{PNASoneF}
\usepackage{PNAStwoF}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% ADDITIONAL OPTIONAL STYLE FILES

%% The AMS math files are commonly used to gain access to useful features
%% like extended math fonts and math commands.

\usepackage{amssymb,amsfonts,amsmath}

%\usepackage{subcaption}
\graphicspath{ {paper_figures2/} }

%\usepackage{sansmath}

%\usepackage[font={sf, small}]{caption}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% OPTIONAL MACRO FILES
%% Insert self-defined macros here.
%% \newcommand definitions are recommended; \def definitions are supported

%\newcommand{\mfrac}[2]{\frac{\displaystyle #1}{\displaystyle #2}}
%\def\s{\sigma}

\DeclareMathSizes{9}{8}{7}{7}

\DeclareMathOperator*{\argmin}{arg\,min}

\makeatletter
\newcommand{\customlabel}[2]{%
\protected@write \@auxout {}{\string \newlabel {#1}{{#2}{}}}}
\makeatother

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Don't type in anything in the following section:
%%%%%%%%%%%%
%% For PNAS Only:
\contributor{Submitted to Proceedings
of the National Academy of Sciences of the United States of America}
\url{www.pnas.org/cgi/doi/10.1073/pnas.0709640104}
\copyrightyear{2008}
\issuedate{Issue Date}
\volume{Volume}
\issuenumber{Issue Number}
%%%%%%%%%%%%

\begin{document}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%% For titles, only capitalize the first letter
%% \title{Almost sharp fronts for the surface quasi-geostrophic equation}

\title{Temporal ordering and registration of cross-sectional imaging data}


%% Enter authors via the \author command.
%% Use \affil to define affiliations.
%% (Leave no spaces between author name and \affil command)

%% Note that the \thanks{} command has been disabled in favor of
%% a generic, reserved space for PNAS publication footnotes.

%% \author{<author name>
%% \affil{<number>}{<Institution>}} One number for each institution.
%% The same number should be used for authors that
%% are affiliated with the same institution, after the first time
%% only the number is needed, ie, \affil{number}{text}, \affil{number}{}
%% Then, before last author ...
%% \and
%% \author{<author name>
%% \affil{<number>}{}}

%% For example, assuming Garcia and Sonnery are both affiliated with
%% Universidad de Murcia:
%% \author{Roberta Graff\affil{1}{University of Cambridge, Cambridge,
%% United Kingdom},
%% Javier de Ruiz Garcia\affil{2}{Universidad de Murcia, Bioquimica y Biologia
%% Molecular, Murcia, Spain}, \and Franklin Sonnery\affil{2}{}}

\author{Carmeline~J.~Dsilva\affil{1}{Department of Chemical and Biological Engineering, Princeton University, Princeton, New Jersey, USA},
Bomyi~Lim\affil{1}{},
Thomas~J.~Levario\affil{2}{School of Chemical and Biomolecular Engineering, Georgia Institute of Technology, Atlanta, Georgia, USA},
Hang~Lu\affil{2}{},
Amit~Singer\affil{3}{Department of Mathematics, Princeton University, Princeton, New Jersey, USA} \affil{4}{Program in Applied and Computational Mathematics, Princeton University, Princeton, New Jersey, USA},
Stanislav~Y.~Shvartsman\affil{1}{} \affil{5}{Lewis-Sigler Institute for Integrative Genomics, Princeton University, Princeton, New Jersey, USA},
\and
Ioannis~G.~Kevrekidis\affil{1}{} \affil{4}{Program in Applied and Computational Mathematics, Princeton University, Princeton, New Jersey, USA}}

\contributor{Submitted to Proceedings of the National Academy of Sciences
of the United States of America}

%% The \maketitle command is necessary to build the title page.
\maketitle

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{article}

\begin{abstract}
In studies of development, researchers are often presented with cross-sectional data, where each data point is a sample from a population fixed at a slightly different developmental time.
%
The goal is then to temporally order the data to reconstruct the developmental dynamics.
%
If each data point is a two-dimensional image, the images must first be registered before they can be temporally ordered.
%
When such data sets are large, noisy, and/or if the developmental changes are subtle, these tasks can be difficult to do by hand.
%
We present an automatic approach to register {\it and} temporally order cross-sectional data sets of images.
%
The mathematical techniques (angular synchronization for image registration, diffusion maps for temporal ordering, and
vector diffusion maps for simultaneously performing both tasks) are applicable to a wide variety of data sets and
require little {\it a priori} knowledge of the image features or the developmental dynamics.
%
We demonstrate the utility of our methods using a collection of images from a study of {\it Drosophila} embryogenesis.
\end{abstract}


%% When adding keywords, separate each term with a straight line: |
\keywords{temporal ordering | image registration}

%% Optional for entering abbreviations, separate the abbreviation from
%% its definition with a comma, separate each pair with a semicolon:
%% for example:
%% \abbreviations{SAM, self-assembled monolayer; OTS,
%% octadecyltrichlorosilane}

% \abbreviations{}

%% The first letter of the article should be drop cap: \dropcap{}
%\dropcap{I}n this article we study the evolution of ''almost-sharp'' fronts

%% Enter the text of your article beginning here and ending before
%% \begin{acknowledgements}
%% Section head commands for your reference:
%% \section{}
%% \subsection{}
%% \subsubsection{}



\dropcap{E}xperimental studies of developmental dynamics fall in two broadly defined categories: longitudinal and cross-sectional \cite{diggle2002analysis}.
%
In longitudinal studies, developmental progress is monitored over time for the same embryo.
%
In a cross-sectional study, one embryo contributes only a single snapshot of a chemical or morphological process along its developmental trajectory, and the developmental dynamics must be reconstructed from multiple snapshots of different embryos.
%
Both of these sampling schemes have their advantages and limitations, and both are extensively used by developmental biologists.
%
Here we focus on cross-sectional studies, which have a time-honored history and still present the only option for most organisms.
%
In a typical cross-sectional study, a group of developing embryos is fixed using a procedure that arrests their development and stained with chemicals that help visualize a handful of cellular processes.
%
Fixed embryos are then imaged using any given number of microscopy techniques.
%
Recent advances in large-scale physical manipulation and imaging of embryos have produced rapidly increasing volumes of cross-sectional data, in which every embryo is observed at a different geometric orientation and developmental time point.
%
Importantly, the ``age'' of any given embryo arrested in its development is not known to high accuracy.
%
In general, it is only known that a collection of embryos belongs to a certain time window.
%
In order to recover the developmental dynamics from such image datasets, snapshots of different embryos must be spatially aligned or registered to factor out the relevant symmetries (i.e., translations and rotations), and then ordered in time.
%
We show how existing dimensionality reduction algorithms can greatly accelerate -and combine- both of these tasks.

Temporal ordering and registration of images is straightforward when the number of images is small and differences between them are visually apparent.
%
As an example, Figure~\ref{fig:fish} shows a caricature of fish development, combining the processes of growth and color patterning.
%
In this case, temporal ordering can be accomplished by arranging the fish by size, which is monotonic with the developmental progress.
%
Image registration is based on obvious morphological landmarks, such as the position of head and fins.
%
On the other hand, real data poses nontrivial challenges, due to the number of images, measurement noise and variability, and the subtlety of the developmental changes.

\begin{figure}[t]
\includegraphics[width=8.4cm]{fig1}
\caption{{\it (A)} Fish, each in a different orientation and a different stage of development. {\it (B)} Fish, now rotationally registered and temporally ordered. For this caricature, the registration and ordering is easy to do ``by hand" because the data set is small and the developmental changes are easy to recognize.}
%\label{fig:fish}
\customlabel{fig:fish}{1}
\customlabel{subfig:fish_unordered}{\ref{fig:fish}{\it A}}
\customlabel{subfig:fish_ordered}{\ref{fig:fish}{\it B}}
\end{figure}

We will demonstrate our techniques using two data sets from a study of {\it Dropophila} embryogenesis.
%
The first data set, shown in Figure~\ref{subfig:raw_data1} is relatively simple and will allow us to both illustrate and validate our methods. 
%
In the second data set, shown in Figure~\ref{subfig:raw_data2}, the dynamics are significantly more complex, and we will show that we can recover a reasonable trajectory that is qualitatively consistent with current knowledge.

In both of these data sets, each image is taken from a different embryo, fixed at a different developmental time and imaged in a different rotational orientation.
%
We would like to organize these data sets in a meaningful and informative way.
%
This requires rotating and translating each image so the set is in a consistent frame of reference ({\it registration}), and then organizing the images. 
%
We will show that the algorithms can effectively order the images in time and therefore give us meaningful picture of the underlying developmental dynamics. 


\begin{itemize}
\item Imaging data is becoming increasingly prevalent in biology research.
%
\item General motivation about biology data ...
%
\item Need methods to organize data, visualize, ...
%
\item REDUCTION is important, reduction because of symmetry and reduction because of content
%
\item There are a bunch of recent techniques- some established (like dmaps), some very fresh, like synchronization or VDMaps,  that have had their own reasons to be developed but they are as if tailor made for what we would want to do
\end{itemize}


\section{Results and Discussion}

%Rank corr for data set 1 using VDM=  0.9127

%Variance captured by PC 1 for data set 2 = 0.2155

\subsection{Vector diffusion maps for registration and temporal ordering}

We use vector diffusion maps \cite{singer2012vector} to automatically register and order our images.
%
Vector diffusion maps combines two algorithms, angular synchroniation \cite{singer2011angular} for registeratin and diffusion maps \cite{coifman2005geometric} for extracting low-dimensional structure, into one computation that allows us to simulatenously register and temporally order our images. 

Registration via angular synchronizatoin/vector diffusion maps uses pairwise alignment information about the images to globally register the images in a consistent way.
%
A schematic illustation is shown in Figure~\ref{subfig:synchron1}, where each image is depicted as a vector, and we would like to align the set of vectors. 
%
We first compute the angles needed to align pairs of vectors (or images).%
This is rather simple to do, and requires no template function, but can be noisy and some images may be misaligned.
%
Using the angles between all pairs of vectors (or images), angular synchronization/ vector diffusion maps then finds the best rotation angle for each vector that is most consistent with the pairwise measurements.

Diffusion maps \cite{coifman2005geometric} uncovers a parameterization of data that lies on a low-dimensional (perhaps nonlinear) manifold in high-diemnsional space. 
%
This is illustrated in Figure~\ref{subfig:dmaps1}, where the data are two-dimensional points which lie on a one-diemsnional nonlinear curve. 
%
The color in Figure~\ref{subfig:dmaps2} depicts the one-dimensional parameterization or ordering of the data that we can detect visually.
%
In our examples, each data point will be an image with many pixels, and so extracting the low-dimensional structure visually will be difficult.
%
Instead, we will rely on diffusion maps to automatically extract this structure from our high-dimensional data.

Both of these steps are accomplished simulataneously using vector diffusion maps.
%
Vector diffusion maps finds a good parameterization of the data, while simultaneously registering data points which are close.

\begin{figure}[t]
\includegraphics[width=8.4cm]{fig4}
\caption{ {\it (A)} Set of vectors, each in a different orientation. The pariwise alignment angles are indicated. {\it (B)} The vectors from {\it A}, each rotated so that the set is globally aligned. Note that the chosen rotation angles are consistent with the pairwise alignments in {\it A}. {\it (C)} Data points (in black) which lie on a one-dimensional nonlinear curve in two dimensions. Each pair of data points is connected by an edge, and the edge weight is related to the Euclidean distance between the points through the diffusion kernel (see {\it SI Appendix}), so that close data points are connected by dark (``stronger'') edges. {\it (D)} The data in {\it C}, colored by the first (non-trivial) eigenvector from the diffusion map computational procedure, $\phi_2$. The color intensity is monotonic with the arclength, thus parameterizing the curve.}
%\label{fig:schematics}
\customlabel{fig:schematics}{4}
\customlabel{subfig:synch1}{\ref{fig:schematics}{\it A}}
\customlabel{subfig:synch2}{\ref{fig:schematics}{\it B}}
\customlabel{subfig:dmaps1}{\ref{fig:schematics}{\it C}}
\customlabel{subfig:dmaps2}{\ref{fig:schematics}{\it D}}
\end{figure}


\subsection{Stage 5 images in {\it Drosophila} embryogenesis}

Our first data set consists of $49$ images from stage 5 of {\em Drosophila} embryogenesis, shown in Figure~\ref{subfig:raw_data1}. 
%
Each image is from a different embryo in a different orientation and at a different developmental time. 
%
We registered and ordered the images using vector diffusion maps.
%
The results are shown in Figure~\ref{subfig:ordered_data1}.

Furthermore, we can gain a better understanding of the developmental trajectory by looking at an averaged trajectroy.
%
Figure~\ref{subfig:average_data1} shows the moving average of the trajectory in Figure~\ref{subfig:ordered_data1} at different points. 
%
We can easily see the developmental dynamics.

Furthermore, for this data set, we can validate our proposed ordering of the data.
%
During stage 5, we can measure the developmental time based on the monotonic progress of cellularization.
%
We can compare this temporal ordering with the ordering we extract from vector diffusion maps. 
%
Quantitatively, the rank correlation coefficient between these two orderings is 0.9127.

\begin{figure*}[t]
\raisebox{8.1cm}{{\figtextfont A}}
\includegraphics[width=8.4cm]{raw_data1}
\hfill
\raisebox{8.1cm}{{\figtextfont B}}
\includegraphics[width=8.4cm]{VDM_data1_ordered}\\
\vspace{0.2cm}
\centering
\raisebox{0.5cm}{{\figtextfont C}}
\includegraphics[width=8.4cm]{average_trajectory_VDM}
\caption{{\it (A)} Images from stage 5 of {\it Drosophila} embryogenesis. Each image is of a different embryo in a different rotational orientation. {\it (B)} Images from {\it A}, now registered and ordered using vector diffusion maps. {\it (C)} Average trajectory of the images shown in {\it B}.}
\customlabel{fig:data1}{3}
\customlabel{subfig:raw_data1}{\ref{fig:data1}{\it A}}
\customlabel{subfig:ordered_data1}{\ref{fig:data1}{\it B}}
\customlabel{subfig:averaged_data1}{\ref{fig:data1}{\it C}}
\end{figure*}


\subsection{Stage 5--7 Images of {\it Drosophila} embryogenesis}

The images in Figure~\ref{fig:data1} are relatively simple, and the methods presented here are perhaps an overkill for these images.
%
By visual inspection, these data are such that principal component analysis \cite{shlens2005tutorial} could uncover meaningful structure, and projection onto the first principal component would be sufficient to order the data.
%
However, often, developmental data are significantly more complex, and nonlinear techniques such as diffusion maps/vector diffusion maps are required to extract meaningful structure in the data. 
%
Figure~\ref{subfig:raw_data2} shows a data set of images from stages 5--7.
%
Only 22\% of the variability in the data is captured by the first principal component, and so we are forced to use more sophisticated techniques. 

Figure~\ref{subfig:ordered_data2} shows the images in Figure~\ref{subfig:raw_data2}, now registered and ordered using vector diffusion maps \cite{singer2012vector}.
%
Unlike the images in Figure~\ref{subfig:raw_data1}, 
for Figure~\ref{fig:data2}, we have no time marker with which to validate our orderings.
%
However, there are noticeable features in the ordering that we can see are consistent with previously known dynamics,
and we can construct a smoothed trajectory by averaging groups of neighboring images.
%
Figure~\ref{subfig:averaged_trajectory} shows the images from Figure~\ref{subfig:ordered_data2}, divided into 12 groups and averaged. 
%
We can easily see the formation of the ventral furrow and germ band elongation, which concludes with ventral cells shifted to the dorsal side of the embryo.

\newpage
\begin{figure*}[t]
\raisebox{5.3cm}{{\figtextfont A}}
\includegraphics[width=16.8cm]{raw_data2}

\vspace{0.2cm}
\raisebox{5.3cm}{{\figtextfont B}}
\includegraphics[width=16.8cm]{VDM_ordered}

\vspace{0.2cm}
\raisebox{0.6cm}{{\figtextfont C}}
\includegraphics[width=16.8cm]{average_trajectory}
\caption{{\it (A)} Images from stages 5-7 of {\it Drosophila} embryogenesis. Each image is of a different embryo in a different rotational orientation. {\it(B)} Data from {\it A}, registred and ordered using vector diffusion maps. {\it (C)} Average trajectory of the images shown in {\it B}.}
\customlabel{fig:data2}{5}
\customlabel{subfig:raw_data2}{\ref{fig:data2}{\it A}}
\customlabel{subfig:ordered_data2}{\ref{fig:data2}{\it B}}
\customlabel{subfig:averaged_data2}{\ref{fig:data2}{\it C}}
\end{figure*}

\section{Conclusions}

We present algorithms for registration and temporal ordering of images.
%
The algorithms are sufficiently general that they can be applied to a wide variety of biological imaging data.

We have shown how symmetry and dimensionality reduction techniques can successfully register and order cross-sectional developmental imaging data.
%
Although we illustrated our methods using a specific data set from a study of {\it Drosophila} embryogenesis, the framework is quite general and requires little knowledge of the image features or the developmental dynamics.
%
To our knowledge, temporal ordering of {\it Drosophila} images has not been automated to this extent.
%
Recently, machine learning algorithms have been used to group {\it Drosophila} images into different developmental classes \cite{yuan2014automated}, yet there, the appropriate classes must be defined by the user and hand-labeled training data are required.

Temporal ordering of cross-sectional biological data has been studied in a variety of other contexts.
%
Cross-sectional RNA sequencing data has been temporally ordered \cite{anavy2014blind, trapnell2014dynamics}, where each data point is the transcriptome for a single sample.
%
Gene expression data from multiple patients have been studied in \cite{gupta2008extracting, qiu2011discovering};
here, each data point is a vector containing the expression levels of the various genes.
%
Snapshots of cells throughout the cell cycle can be described by a vector of features (e.g., amount of DNA) which quantify the cell's state  \cite{kafri2013dynamics}.
%
These problems are no different than our task of ordering images, each of which is a large vector of pixel intensities.
%
The majority of these works temporally order the data either by solving a traveling salesman problem on the data, or by constructing a minimum spanning tree of the data,
expecting that these structures (the path of the traveling salesman, or the minimum spanning tree) characterize the majority of the structure/variability within the data and provide an accurate encoding of the dynamical progression.
%
We, instead, use diffusion maps to construct a one-dimensional parameterization of the data.
%
Diffusion maps are one of many nonlinear dimensionality reduction techniques that have been recently developed \cite{Belkin2003, tenenbaum2000global, Donoho2003, Roweis2000}.
%
They have been shown to be more robust to noise than other path-based algorithms, such as isomap \cite{balasubramanian2002isomap}, and so we suspect they may perform better than the ordering algorithms used in previous work.
%
{\bf Yannis says ``careful'', he thinks this might be too strong of a statement.}
%%%
%%%   we need help here
%%%
%
%Furthermore, although we use only the first diffusion maps coordinate to order the data (we assume the data are one-dimensional), diffusion maps are not limited to one-dimensional data and can be used to extract more complex structures.
%
%For example, in data which contains several branches or classes, the first few coordinates will delineate the branches or separate the classes.

The task of image registration has been widely studied \cite{zitova2003image}, for applications such as face recognition \cite{rowley1998rotation}, medical image registration \cite{hajnal2010medical}, and texture classification \cite{greenspan1994rotation}.
%
The main point to note here is that many registration algorithms require definition of appropriate landmarks or features for the images and the location of these within each image.
%
For example, in face recognition, the first step is often to find the eyes, nose, and mouth within the face, and then register images by aligning these features \cite{zhao2003face}.
%
Temporally ordering images of a construction site over time is also based on first identifying features within the images \cite{schindler2007inferring}.
%
In general, appropriate features may not be known {\it a priori}, and so we require a registration algorithm that does not rely on predefined features, but only on the inherent geometric symmetry of the problem.
%
Template-based methods are also common for registration, where one optimally aligns each image to a predefined template function.
%
However, these methods can suffer when the images are noisy, and the success of such methods can depend strongly on the template function \cite{shatsky2009method}.
%
{\bf Need to discuss consistency.}

Angular synchronization and vector diffusion maps require no feature points and no template function.
%
These methods have been used to register two-dimensional cryo-electron microscopy images of three-dimensional molecules \cite{singer2011three}, where each image is a picture of (essentially) the {\it same} molecule; variation in the data comes from differences in the viewing angles (rotations of the molecule) and from instrument/measurement noise.
%
In contrast, our data contains variation from the different embryo orientations, as well as variation from the underlying developmental dynamics. % (which underpins the development of vector diffusion map algorithms).
%
In addition to the eigenproblem relaxation used here, other synchronization algorithms have been proposed \cite{bandeira2014multireference}.
%
Synchronization is formulated for general symmetry groups, and depending on the data, one might consider other symmetries, such as reflections and scaling/dilation.
%
Here, the main requirement is that the symmetry group be represented by orthogonal or unitary matrices \cite{singer2013spectral}.

Temporal ordering can also be accomplished without first registering the images, by constructing functions of the images which are invariant under translations and rotations. 
%
These functions are then used as the data in the diffusion maps calculation; since these representative functions do not change under translations and rotations, the resulting diffusion map embedding is invariant to symmetries.
%
Such functions include the scale invariant feature transform \cite{lowe1999object, lowe2004distinctive}, the scattering transform \cite{mallat2012group}, and the bispectrum \cite{zhao2014rotationally}.
%
These methods have the disadvantage that we do not obtain a registered set of images, but only a temporal ordering.
%
However, such methods can often be faster for large data sets, since the functions are computed per image, rather than computing pairwise quantities in synchronization.

In all of these data-driven methods, an essential question is how much and what type of data is needed to accurately reconstruct the dynamics.
%
For diffusion maps to temporally order the data, we require the developmental trajectory to be well-sampled, and that the principal direction of variability in the data (the coordinate which diffusion maps will recover) be monotonic in time.
%
In addition, the data should have spatial variability/structure for synchronization to successfully register the images.
%
In our example, the data set is sufficiently large so that the dynamics are well-sampled.
%
We have included both the red and the green channels so that the principal direction of variability is monotonic in time.
%
If we only used the green channel in our analysis, we could not successfully order the data using diffusion maps, as this signal does not change appreciably in time.
%
Both the red and green channels contain spatial variability which allows us to register the data points: the green channel has one peak at a specific angular around the embryo, and the red channel has two peaks which grow in time.
%
We include the green channel because the location of the red peaks is noisy, and so inclusion of the (less noisy) green channel allows us to more accurately register the images.
%
%For the methods we present here, we require a sufficient amount of data for the one-dimensional evolution curve to be ``seen'' by the algorithms (i.e., in Figure~\ref{subfig:dmaps_edges}, there are a sufficient number of data points so that our eye can see the curve; if there were only 4 data points, it would most likely be difficult for our eye to detect the one-dimensional structure).
%
%We also require the data to contain enough information so that the alignments and dynamics are apparent.
%
%For example, in the data presented in Figure~\ref{subfig:images_unordered}, the green peaks delineate the position of the ventral axis in the embryo, and the intensity of the red peaks grow over time.
%
%Therefore, these images contain information about rotational symmetries of the data as well as the dynamics of the data.
%
%If we removed the green signal, and only used the red signal in our analysis, the registration would not be as accurate.
%
%Furthermore, if we only used the green signal, then there would be no dynamic information (the shape or intensity of Dorsal does not change as a function of time in the developmental stage we are considering), and the temporal ordering would be inaccurate.
%
We are confident that for many applications, these requirements - data which adequately samples the developmental trajectory, and data with sufficient spatial and temporal variability - are easily met.
%
In these cases, our methods allow for the rapid, automated analysis of imaging data sets to help uncover possibly complex dynamics and structure.





Although the examples here focus on temporally ordering the data, tasks such as classification and comparison of different mutations can also be accomplished using the same techniques. 
%

\begin{itemize}
\item What are other people doing  (maybe critically)
\item Movies  smoothness
\item Discrete symmetries like flips
\item Local information ? (features of the image) (associate with Fourier-Bessel and bispectra)
\item Mutant-wildtype stuff  IMPORTANT
\item And in general, MULTIPLE variablities is important (mutants, guys in another cycle, etc.)
 
\item size is an issue
\item  small tilts is an issue
\item   could one do three-d image processing ?  of the floating thing, without the device
\item how do we go about merging datasets ?
\end{itemize}



\bibliographystyle{pnas}
\bibliography{background_reading/references,../../references/references}

\end{article}

\end{document}

